executor/sort.go

```diff
@@ -451,24 +464,25 @@ func (e *TopNExec) executeTopN(ctx context.Context) error {
 	for {
 		err := Next(ctx, e.children[0], childRowChk)
 		if err != nil {
-			return err
+			return false, err
 		}
 		if childRowChk.NumRows() == 0 {
 			break
 		}
 		err = e.processChildChk(childRowChk)
 		if err != nil {
-			return err
+			return false, err
 		}
 		if e.rowChunks.Len() > len(e.rowPtrs)*topNCompactionFactor {
 			err = e.doCompaction()
 			if err != nil {
-				return err
+				return false, err
 			}
+			compacted = true
```

Comment:
What happen if we get more tuples after `doCompaction()`? It seems to be possible to contain tuples with different schema in `e.rowChunks`.

Reply:
Great catch!